src/SFA.DAS.AssessorService.Data/CertificateRepository.cs

```diff
@@ -300,40 +300,129 @@ public async Task UpdateCertificatesReadyToPrintInBatch(Guid[] certificateIds, i
                 transaction: _unitOfWork.Transaction);
 
             await AddMultipleCertificateLogs(certificateIds, CertificateActions.Status, null, null, null, SystemUsers.PrintFunction, batchNumber, null);
-        }        
-                
-        public async Task<PaginatedList<Certificate>> GetCertificateHistory(string endPointAssessorOrganisationId, int pageIndex, int pageSize, List<string> statuses)
-        {
-            var count = await GetCertificatesCount(endPointAssessorOrganisationId, statuses);
-
-            var ids = await (from certificate in _context.Certificates
-                             join organisation in _context.Organisations on
-                               certificate.OrganisationId equals organisation.Id
-                             where organisation.EndPointAssessorOrganisationId == endPointAssessorOrganisationId
-                               && !statuses.Contains(certificate.Status)
-                             orderby certificate.CreatedAt descending
-                             select certificate.Id)
-                                        .Skip((pageIndex - 1) * pageSize)
-                                        .Take(pageSize).ToListAsync();
-
-            var certificates = await _context.Certificates.Where(q => ids.Contains(q.Id))
-                .Include(q => q.Organisation)
+        }
+
+        public async Task<PaginatedList<Certificate>> GetCertificateHistory(string endPointAssessorOrganisationId,
+                int pageIndex,
+                int pageSize,
+                string searchTerm,
+                string sortColumn,
+                bool sortDescending,
+                List<string> statuses)
+        {
+
+            var certificatesQuery = (from certificate in _context.Certificates
+                 .Include(q => q.Organisation)
```

Comment:
Why is Organisation included here and then also joined at line 318? and wouldn't it read better to just use all join syntax statements?

Reply:
coming back to this one, does EF not generate the joins based on the Include()? Could this not all be achieved with a LINQ where and  move the JSON serialization within the Select()?

`var certificatesQuery = _context.Certificates
                .Include(c => c.Organisation)
                .Include(c => c.CertificateLogs)
                .Include(c => c.CertificateBatchLog)
                .Where(c => c.Organisation.EndPointAssessorOrganisationId == endPointAssessorOrganisationId && !statuses.Contains(c.Status))
                .Select(c => new SortableCertificateInfo
                {
                    Certificate = c,
                    Organisation = c.Organisation,
                    CertificateData = JsonConvert.DeserializeObject<CertificateData>(c.CertificateData)
                });`

Reply:
Yes but I have a more simplex version with  out .Deserialization out of SQL